### PR TITLE
fix ordering of quantiles in describe.

### PR DESCRIPTION
**Pull request by gregrf**
Previously dataframe.describe was using the built-in set to sort percentiles which doesn't sort properly. This led to unpredictably wrong results. For example percentiles=[0.25, 0.5, 0.75, 0.99] was reordered by set to [0.25, 0.5, 0.99, 0.75] which broke dataframe.quantiles which assumes quantiles are sorted by value.

After this patch the built-in sorted will be used to sort quantiles.

- [x] Tests added / passed
- [x] Passes `flake8 dask`

### CODE REVIEW

**Comment by martindurant** [comment]
Can you please include a test with your original set of quantiles, so we can prove this is fixed?

Fixes #4642

**Comment by gregrf** [comment]
I added a simple test to catch this error.

**Comment by martindurant** [comment]
Thank you, that's perfect.

**Comment by martindurant** [comment]
Appears to fail on flake8 linting:
```
dask/dataframe/tests/test_dataframe.py:322:1: W293 blank line contains whitespace
dask/dataframe/tests/test_dataframe.py:325:1: W293 blank line contains whitespace
dask/dataframe/tests/test_dataframe.py:331:1: W293 blank line contains whitespace
dask/dataframe/tests/test_dataframe.py:338:1: E302 expected 2 blank lines, found 1
```

**Comment by gregrf** [comment]
Yesterday's patch, https://github.com/dask/dask/pull/4650, should this patch is still important. Currently in `describe`, if the percentiles argument is treated with this line if it is not `None`

```python
percentiles = list(set(sorted(percentiles + [0.5])))
```
But look what happens if `percentiles` is an `ndarray`:

```python
A = dask.array.arange(101)
s = dask.dataframe.from_dask_array(A)
s.describe(percentiles=np.array([0.25])).compute()
```

count    101.000000
mean      50.000000
std       29.300171
min        0.000000
75%       75.000000
max      100.000000
dtype: float64

It's calculated the 75'th% instead of the 25'th% because numpy added [0.5] instead of concatenating like `list` would. I've added tests where `percentiles` is a `list`, `tuple` or `ndarray` to make sure it works for all of those list-like types.

I'm off on vacation without a computer. Back in a week.

**Comment by martindurant** [comment]
Have a nice holiday!
It would be good to have a test for the specific case you cite; although I note that the docstring says the input should be "list-like", which may or may not include array.

**Comment by gregrf** [comment]
Hi, I'm back.

As a user I'm certainly surprised whenever a function in the scipy ecosystem doesn't accept a 1d ndarray when a 1-d container is needed but I admit to having to had to google "list-like." Is there a tight definition for this in dask? It turns out Pandas has a function `pandas.api.types.is_list_like()` that considers ndarrays to be list-like although it also considers the builtin `set` and `pandas.series` to be list-like which surprises me a little. I'm any event, this patch fixes the problem.

File: dask/dataframe/core.py [hunk 1]
```diff
@@ -1543,7 +1543,12 @@ def describe(self, split_every=False, percentiles=None):
         if percentiles is None:
             percentiles = [0.25, 0.5, 0.75]
         else:
-            percentiles = sorted(set(list(percentiles) + [0.5]))
+            # always include the the 50%tle to calculate the median
+            # unique removes duplicates and sorts quantiles
+            percentiles = np.array(percentiles)
+            percentiles = np.append(percentiles, 0.5)
```
Comment (martindurant) [inline dask/dataframe/core.py]: why do we always need the median?

Comment (gregrf) [reply dask/dataframe/core.py]: I don't know. It's just how pandas does it. It is unexpected though so I thought a comment clarifying why the 0.5 comes out of nowhere couldn't hurt.

Comment (martindurant) [reply dask/dataframe/core.py]: Oh indeed, you are right - that's fine then, I suppose.

**Comment by martindurant** [comment]
Failure comes from seemingly unrelated bokeh thing

**Comment by gregrf** [comment]
Sorry, I'm new to this. Am I right that it's passing for PYTHON=3.5 NUMPY=1.11.1 PANDAS=0.19.2 and below and failing for newer packages? I don't know what to make of that.

**Comment by martindurant** [comment]
You do not need to do anything, I don't think this is your fault. @mrocklin @jakirkham , is the bokeh failure a known thing? 
I am +1 on this PR, that aside.

**Comment by martindurant** [comment]
(Bokeh 1.1.0 was released today, with a lot of layout refactoring)

**Comment by mrocklin** [comment]
It looks new to me.  I agree with Martin's assessment that it's probably
due to the 1.1.0 release.

I'm slammed today.  Does anyone else have time to resolve this?

**Comment by martindurant** [comment]
Should be fixed in https://github.com/dask/dask/pull/4680

**Comment by martindurant** [comment]
OK, so merging this now. Thank you @gregrf

**Pull request merged**